server/service/src/invoice_line/inbound_shipment_line/update/generate.rs

```diff
@@ -94,13 +95,14 @@ fn generate_line(
 
     if let Some(total_before_tax) = total_before_tax {
```

Comment:
I would simplify this slightly with:
```rust
  update_line.total_before_tax = if let Some(total_before_tax) = total_before_tax {
        total_before_tax
    } else if number_of_packs.is_some() || cost_price_per_pack.is_some() {
        update_line.cost_price_per_pack * update_line.number_of_packs as f64
    } else {
        update_line.total_after_tax
    };
```
`update_line.total_before_tax =` -> indicates the result of conditions to always be new total_before_tax

`  update_line.cost_price_per_pack * update_line.number_of_packs` -> number of packs and cost_price_per_pack already set to new values.

Reply:
Would the last else statement not be pdate_line.cost_price_per_pack * update_line.number_of_packs as f64? 🤔